tests/cupy_tests/math_tests/test_matmul.py

```diff
@@ -129,6 +129,118 @@ def test_cupy_matmul(self, xp, dtype1, dtype2):
         return xp.matmul(x1, x2)
 
 
+class _TestMatmulComputeTypes(unittest.TestCase):
+
+    def setUp(self):
+        self.old_compute_type = cupy.core.get_compute_type(self.dtype)
+        cupy.core.set_compute_type(self.dtype, self.compute_type)
+
+    def tearDown(self):
+        cupy.core.set_compute_type(self.dtype, self.old_compute_type)
+
+    def make_x1_x2(self, xp, shapes, dtypes):
+        x1 = testing.shaped_random(shapes[0], xp, dtypes[0])
+        x2 = testing.shaped_random(shapes[1], xp, dtypes[1])
+        return x1, x2
+
+
+@testing.parameterize(
+    *testing.product({
+        'compute_type': [
+            cupy.core.core.COMPUTE_TYPE_DEFAULT,
+            cupy.core.core.COMPUTE_TYPE_PEDANTIC,
+        ],
+        'shape_pair': [
+            ((100, 200), (200, 300)),
+            ((200, 300), (300, 100)),
+            ((300, 100), (100, 200)),
+        ],
+    }))
+@testing.gpu
+class TestMatmulFp16ComputeTypes(_TestMatmulComputeTypes):
+    dtype = numpy.float16
+
+    @testing.numpy_cupy_allclose(rtol=1e-3, atol=1e-3)
+    def test_operator_matmul(self, xp):
+        x1, x2 = self.make_x1_x2(xp, self.shape_pair, (self.dtype, self.dtype))
+        return operator.matmul(x1, x2)
+
+    @testing.numpy_cupy_allclose(rtol=1e-3, atol=1e-3)
+    def test_cupy_matmul(self, xp):
+        x1, x2 = self.make_x1_x2(xp, self.shape_pair, (self.dtype, self.dtype))
+        return xp.matmul(x1, x2)
+
+
+@testing.parameterize(
+    *testing.product({
+        'compute_type': [
+            cupy.core.core.COMPUTE_TYPE_DEFAULT,
+            cupy.core.core.COMPUTE_TYPE_PEDANTIC,
+            cupy.core.core.COMPUTE_TYPE_TF32,
+        ],
+        'shape_pair': [
+            ((100, 200), (200, 300)),
+            ((200, 300), (300, 100)),
+            ((300, 100), (100, 200)),
+        ],
+        'dtype_pair': [
+            (numpy.float16, numpy.float32),
+            (numpy.float32, numpy.float32),
+            (numpy.float16, numpy.complex64),
+            (numpy.float32, numpy.complex64),
+            (numpy.complex64, numpy.complex64),
+        ],
+    }))
+@testing.gpu
+class TestMatmulFp32ComputeTypes(_TestMatmulComputeTypes):
+    dtype = numpy.float32
+
+    @testing.numpy_cupy_allclose(rtol=1e-3, atol=1e-3)
```

Comment:
Just to make sure, have you tested this with CC 8.0 and the tolerance is enough, right?

Reply:
I will check it now, so please wait for a while.

Reply:
The tolerance was OK for TF32 on A100, but not OK for FP16..
Let me use smaller matrices for FP16 tests.